### ISSUE TITLE
explicitly-centered levels sometimes are not, as of v6.203

### ISSUE THREAD

**Issue by karlmsmith** [NOAA-PMEL/Ferret#979]
**Reported by andrew wittenberg on 9 Oct 2009 17:45 UTC**
The following bug shows up in v6.203 dated 6/26/09, but didn't in v6.2beta dated 5/19/09.  Levels should be centered here, but an extra level gets tossed in at the high end, throwing off the centering of the other levels.

```
        NOAA/PMEL TMAP
        FERRET v6.203  
        Linux(g77) 2.4.21-32 - 06/26/09
         9-Oct-09 13:38     

yes? use coads_climatology
yes? shade/lev=(-1.2,1.2,.1)/l=1 sst
```

Migrated-From: http://dunkel.pmel.noaa.gov/trac/ferret/ticket/1707

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 9 Oct 2009 18:07 UTC**
See #1613; looks like the same thing.

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 9 Oct 2009 18:59 UTC**
This is a precision bug, occurs only in 32-bit linux version not the 64-bit one.  It's in the code that fixes bug #1672, where we need to specifically compare values only to single-precision.

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 9 Oct 2009 21:21 UTC**
This is fixed, in plot/levset.F

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 5 Nov 2009 23:43 UTC**

Now Ryo Furue reports in
http://www.pmel.noaa.gov/maillists/tmap/ferret_users/fu_2009/msg00740.html

that

```
  yes? set data coads_climatology
  yes? contour/hlimits=160:180/vlimits=40:60/lev=(3,3.9,0.1)/l=1 sst

```

draws a contour at 4.0 though it should stop at 3.9 (version 6.3 Ferret, and it occurs only in the 32-bit version).

**Comment by karlmsmith** [comment]
**Comment by ansley.b.manke on 5 Nov 2009 23:52 UTC**
And, the case reported just above is fixed by the change in levset noted in hte comment on 10/09/09.  This fix was made after the v6.3 release which was on 9/28.

**Comment by karlmsmith** [comment]
**Attachment from andrew.wittenberg on 9 Oct 2009 17:45 UTC**
The resulting plot.
REPLACE THIS TEXT WITH UPLOADED FILE ./attachments/TRAC_1707_GIT_979/plot.gif
